lang/cs/Org.Apache.REEF.Wake/Remote/Impl/TransportServer.cs

```diff
@@ -89,10 +80,45 @@ public IPEndPoint LocalEndpoint
         /// </summary>
         public void Run()
         {
-            _listener.Start();
+            if (LocalEndpoint.Port == 0)
+            {
+                FindAPortAndStartListener();
+            }
+            else
+            {
+                _listener.Start();
```

Comment:
I still feel that we should only assign it once. I won't hold up the pull request for it, though.